pkg/common/config/config.go

```diff
@@ -735,3 +735,31 @@ func GetConfigPath(ctx context.Context) string {
 	}
 	return cfgPath
 }
+
+// GetSessionUserAgent returns clusterwise unique useragent
+func GetSessionUserAgent(ctx context.Context, useragent string) string {
+	log := logger.GetLogger(ctx)
+	if strings.TrimSpace(useragent) == "" {
+		useragent = "k8s-csi-useragent"
+	}
+	clusterFlavor, err := GetClusterFlavor(ctx)
+	if err != nil {
+		log.Errorf("failed retrieving cluster flavor. Error: %v", err)
+		return useragent
```

Comment:
why are we masking error and returning useragent?

Reply:
fixed in latest commit